app/src/conditional_layer.c

```diff
@@ -68,20 +68,40 @@ static void conditional_layer_deactivate(int8_t layer) {
 
 // On layer state changes, examines each conditional layer config to determine if then-layer in the
 // config should activate based on the currently active set of if-layers.
-static int layer_state_changed_listener(const zmk_event_t *ev) {
+static void update_layer_state_work_handler(struct k_work *work) {
+    int8_t max_then_layer = -1;
+    uint32_t then_layers = 0;
+    uint32_t then_layer_state = 0;
     for (int i = 0; i < NUM_CONDITIONAL_LAYER_CFGS; i++) {
         const struct conditional_layer_cfg *cfg = CONDITIONAL_LAYER_CFGS + i;
         zmk_keymap_layers_state_t mask = cfg->if_layers_state_mask;
+        then_layers |= BIT(cfg->then_layer);
+        max_then_layer = MAX(max_then_layer, cfg->then_layer);
 
```

Comment:
I feel like there's probably a macro-y way to compute `then_layers` and `max_then_layer` as compile-time constants, but maybe that's overkill?

Reply:
I'd say overkill for what we're doing.